File: 백준/Gold/11729. 하노이 탑 이동 순서/하노이 탑 이동 순서.cc
#include <iostream>
#include <math.h>
using namespace std;


void hanoi(int a, int b, int n) {
	if (n == 1) {
		cout << a << ' ' << b << '\n';
		return;
	}
	hanoi(a, 6 - a - b, n - 1);
	cout << a << ' ' << b << '\n';
	hanoi(6 - a - b, b, n - 1);
}

int main() {
	ios::sync_with_stdio(0);
	cin.tie(0);
	int k;
	cin >> k;
	cout << int(pow(2, k) - 1) << '\n';
	hanoi(1, 3, k);
}

File: 백준/Gold/11729. 하노이 탑 이동 순서/하노이 탑 이동 순서.py
def hanoi(a, b, n):
    if n == 1:
        print(f"{a} {b}")
        return
    hanoi(a, 6 - a - b, n - 1)
    print(f"{a} {b}")
    hanoi(6 - a - b, b, n - 1)

k = int(input())
print(2 ** k - 1)
hanoi(1, 3, k)
